manifests/server/config.pp

```diff
@@ -53,12 +53,33 @@
   $ssl             = $mongodb::server::ssl
   $ssl_key         = $mongodb::server::ssl_key
   $ssl_ca          = $mongodb::server::ssl_ca
+  $package_name    = $mongodb::server::package_name
+  $options         = $mongodb::server::options
 
   File {
     owner => $user,
     group => $group,
   }
 
+  file { '/etc/sysconfig/mongod':
+    content => template('mongodb/mongod.erb'),
+    mode    => '0644',
+    owner   => 'root',
+    group   => 'root',
+    notify  => Class['mongodb::server::service']
+  }
+
+  if ($logpath) {
+    file { $logpath:
+      ensure  => present,
+      recurse => true,
+      mode    => '0755',
+      owner   => $user,
+      group   => $group,
+      require => Package[$package_name]
```

Comment:
This order should already be enforced by the class-level dependencies in server.pp